concept/type_/type_manager.rs

```diff
@@ -1692,27 +1885,46 @@ impl TypeManager {
         self.unset_annotation::<AnnotationAbstract>(snapshot, type_, AnnotationCategory::Abstract)
     }
 
-    pub(crate) fn unset_owner_annotation_abstract(
+    pub(crate) fn unset_relation_type_annotation_abstract(
+        &self,
+        snapshot: &mut impl WritableSnapshot,
+        relation_type: RelationType<'static>,
+    ) -> Result<(), ConceptWriteError> {
+        OperationTimeValidation::validate_no_capabilities_with_abstract_interfaces_to_unset_abstractness::<
+            Relates<'static>,
+        >(snapshot, self, relation_type.clone())
+        .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
+
+        self.unset_object_type_annotation_abstract(snapshot, relation_type)
+    }
+
+    pub(crate) fn unset_object_type_annotation_abstract(
         &self,
         snapshot: &mut impl WritableSnapshot,
         type_: impl ObjectTypeAPI<'static> + KindAPI<'static>,
     ) -> Result<(), ConceptWriteError> {
-        // TODO: We may move it to schema validation
-        OperationTimeValidation::validate_no_abstract_attribute_types_owned_to_unset_abstractness(
+        OperationTimeValidation::validate_no_capabilities_with_abstract_interfaces_to_unset_abstractness::<Owns<'static>>(
             snapshot,
+            self,
             type_.clone(),
         )
         .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
 
-        Self::unset_annotation_abstract(self, snapshot, type_)
+        OperationTimeValidation::validate_no_capabilities_with_abstract_interfaces_to_unset_abstractness::<
+            Plays<'static>,
+        >(snapshot, self, type_.clone())
+        .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
+
+        self.unset_annotation_abstract(snapshot, type_)
     }
 
     pub(crate) fn unset_attribute_type_annotation_abstract(
         &self,
         snapshot: &mut impl WritableSnapshot,
         type_: AttributeType<'static>,
     ) -> Result<(), ConceptWriteError> {
-        OperationTimeValidation::validate_no_subtypes_for_type_deletion(snapshot, type_.clone())
+        // TODO: It can be ignored for operation time validations, but we may be strict as well
```

Comment:
I feel like it's ok to be stricter for attribute types as you can't have a non-abstract attribute supertype! But maybe it's too much and a commit time validation is enough

Reply:
I think doing it and being stricter is ok!

Reply:
Removing the todo